pkg/tfgen/docs.go

```diff
@@ -1129,6 +1151,117 @@ func (p *tfMarkdownParser) parseImports(subsection []string) {
 	}
 }
 
+// Recognizes import sections such as ones found in aws_accessanalyzer_analyzer. If the section is
+// recognized, patches up instructoins to make sense for the Pulumi projection.
+func tryParseV2Imports(typeToken string, markdownLines []string) (string, bool) {
+	var out bytes.Buffer
+	fmt.Fprintf(&out, "## Import\n\n")
+
+	markdown := strings.Join(markdownLines, "\n")
+	pn := parseNode(markdown)
+	if pn == nil {
+		return "", false
+	}
+
+	foundCode := false
+
+	for {
+		recognized := false
+		switch pn.Type {
+		case bf.CodeBlock:
+			code := string(pn.Literal)
+			switch string(pn.CodeBlockData.Info) {
+			case "terraform":
+				// Ignore terraform blocks such as:
+				//
+				// ```terraform
+				// import {
+				// 	to = aws_accessanalyzer_analyzer.example
+				// 	id = "example"
+				// }
+				// ```
+				recognized = true
+			case "console", "":
+				// Recognize import example codeblocks.
+				if parsed, ok := parseImportCode(code); ok {
+					emitImportCodeBlock(&out, typeToken, parsed.Name, parsed.ID)
+					recognized = ok
+					foundCode = true
+				}
+			}
+		case bf.Heading:
+			if pn.FirstChild != nil && pn.FirstChild.Type == bf.Text {
+				if string(pn.FirstChild.Literal) == "Import" {
+					// Skip "## Import" heading.
+					recognized = true
+				}
+			}
+		case bf.Paragraph:
+			// Propagate paragraphs to output.
+			paraMD, err := parseTextSeq(pn.FirstChild, false)
+			if err == nil {
+				fmt.Fprintf(&out, "%s\n\n", paraMD)
+				recognized = true
+			}
+		}
+		if !recognized {
+			return "", false
+		}
+		pn = pn.Next
+		if pn == nil {
+			break
+		}
+	}
+	if !foundCode {
+		return "", false
+	}
+	return out.String(), true
+}
+
+func emitImportCodeBlock(w io.Writer, typeToken, name, id string) {
+	fmt.Fprintf(w, "```sh<break>\n")
+	fmt.Fprintf(w, "$ pulumi import %s %s %s\n", typeToken, name, id)
+	fmt.Fprintf(w, "<break>```<break>\n")
+}
+
+var importCodePattern = regexp.MustCompile(
+	`^[%$] (?:pulumi|terraform) import` +
+		`[\\\s]+([^.]+)[.]([^\s]+)[\\\s]+([^\s]+)\s*$`)
+
+// Data correction: ignore the first of two acidentally repeated tokens such as:
+//
+//	% pulumi import aws_n_policy.example aws_n_policy.example id
+var importCodePattern2 = regexp.MustCompile(
+	`^[%$] (?:pulumi|terraform) import[\\\s]+[^\s]+` +
+		`[\\\s]+([^.]+)[.]([^\s]+)[\\\s]+([^\s]+)\s*$`)
+
+// Recognize import example codeblocks.
+//
+// Example:
+//
+//	s := "% pulumi import aws_accessanalyzer_analyzer.example exampleID"
+//	v, ok := parseImportCode(s)
+//	v.Name == "example"
+//	v.ID == "exampleID"
+func parseImportCode(code string) (struct {
+	Name string
+	ID   string
+}, bool) {
+	type ret struct {
+		Name string
+		ID   string
+	}
+	if importCodePattern.MatchString(code) {
+		matches := importCodePattern.FindStringSubmatch(code)
+		return ret{Name: matches[2], ID: matches[3]}, true
+	}
+	if importCodePattern2.MatchString(code) {
```

Comment:
This may be a bit paranoid, but when pattern 2 matches we don't actually know that it was due to accidentally repeated tokens. The additional string could be different from the matched one.

Reply:
moved it to https://github.com/pulumi/pulumi-aws/issues/2760